dotnet/sdk-phase-1.md

```diff
@@ -0,0 +1,1022 @@
+  classes
+  * Some of the options things like activities method references can't by loaded from a file, so we should take a look
+    at helpers for converting activity strings to activity classes reflectively
+  * We need a sample showing configuring workers via config file akin to Spring Boot
+* See the host section for how it all may come together inside a worker service
+
+### Logging
+
+https://learn.microsoft.com/en-us/dotnet/core/extensions/logging
+
+* Unfortunately there is no logging in the standard library for .Net, therefore in order to support this we'd have to
+  add a dependency on https://www.nuget.org/packages/Microsoft.Extensions.Logging
+* NuGet does not support optional dependencies
+* Should we add the dependency?
+  * For:
+    * It's pretty common, e.g. Azure durable task framework has it (well, `Microsoft.Extensions.Logging.Abstractions`)
+      but a really old version from 4 years ago
+  * Against:
+    * We don't like extra dependencies, even these tiny ones
+    * We will have a rudimentary `Temporal.ITemporalLogger` with a simple console implementation by default
+      * Could have a separate `Temporal.Extensions` project that had a `ITemporalLogger` impl for `ILogger`, but this is
+        so simple it can just be a sample
+  * Leaning towards "Against"
+
+### HostBuilder and Worker Services
+
+https://learn.microsoft.com/en-us/dotnet/core/extensions/generic-host and
+https://learn.microsoft.com/en-us/dotnet/core/extensions/workers
+
+* Same discussion around dependencies as logging, this would need `Microsoft.Extensions.Hosting`
+* Our `Temporal.Worker.TemporalWorker` is not a subclass of `BackgroundService` to avoid the dependency
+  * `TemporalWorker.ExecuteAsync` matches `BackgroundService.ExecuteAsync` very intentionally, so this is incredibly
+    easy for a user to create and we will of course have a sample
+
+## General Decision Justifications/Discussion
+
+#### Why `Temporalio` NuGet package but `Temporal` namespace?
+
+* Does not matter. We should discuss and if we prefer our namespace to match our NuGet package, we should change to
+`Temporalio`.
+* `Temporal` is already taken on NuGet. But `Temporal.Sdk` is not, but that could cause confusion with `Temporal`
+  * We already need to apply for https://learn.microsoft.com/en-us/nuget/nuget-org/id-prefix-reservation
+  * Maybe we can persuade https://github.com/ritterim/temporal to deprecate and
+    https://learn.microsoft.com/en-us/nuget/nuget-org/deprecate-packages#transfer-popularity-to-a-newer-package but it
+    may require some ownership work
+
+#### Why `Temporalio`/`Temporal` NuGet/namespace and not `Temporalio.Sdk`/`Temporal.Sdk` NuGet/namespace?
+
+* Seemed simpler at first to just have `Temporal`
+* After more thought, maybe `Temporal.Sdk` would be better? It'd conform better with
+  https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-namespaces
```

Comment:
I'd definitely do this one

Reply:
I'm starting to lean this way too. I think `Temporal.Sdk` _does_ sound better. But then I would want `Temporalio.Sdk` NuGet package? Technically the NuGet package and the actual in-use namespace can be different, and it happens plenty, but most like it the same.

Maybe I should just go ahead and use `Temporalio.Sdk` for the namespace and NuGet package? `Temporalio` isn't too ugly as a root namespace right?